### ISSUE TITLE
Fresh Ubuntu install broken

### ISSUE THREAD

**Issue by maxgittelman** [afrl-rq/OpenUxAS-SoI#44]
I am freshly installing this project and in Ubuntu 16.04 and encountered a build error

```
~/UxAS_pulls/OpenUxAS $ meson build --buildtype=release
The Meson build system
Version: 0.43.0
Source dir: /home/maxgittelman/UxAS_pulls/OpenUxAS
Build dir: /home/maxgittelman/UxAS_pulls/OpenUxAS/build
Build type: native build
Project name: UxAS
Native C compiler: cc (gcc 5.4.0)
Native C++ compiler: c++ (gcc 5.4.0)
Build machine cpu family: x86_64
Build machine cpu: x86_64
Using zeromq from cache.
Using zeromq-4.1.6_patch.tar from cache.

Executing subproject zeromq.

Project name: zeromq
Native C++ compiler: c++ (gcc 5.4.0)
Has header "alloca.h": YES
Has header "arpa/inet.h": YES
Has header "dlfcn.h": YES
Has header "errno.h": YES
Has header "ifaddrs.h": YES
Has header "inttypes.h": YES
Has header "limits.h": YES
Has header "memory.h": YES
Has header "netinet/in.h": YES
Has header "netinet/tcp.h": YES
Has header "stdbool.h": YES
Has header "stddef.h": YES
Has header "stdint.h": YES
Has header "stdlib.h": YES
Has header "strings.h": YES
Has header "string.h": YES
Has header "sys/eventfd.h": YES
Has header "sys/socket.h": YES
Has header "sys/stat.h": YES
Has header "sys/time.h": YES
Has header "sys/types.h": YES
Has header "sys/uio.h": YES
Has header "time.h": YES
Has header "unistd.h": YES
Has header "windows.h": NO
Has header "sys/uio.h": YES
Checking if "kqueue links" links: NO
Checking if "epoll runs" runs: YES
Checking for function "clock_gettime": YES
Checking for function "fork": YES
Checking for function "gettimeofday": YES
Checking for function "gss_init_sec_context": NO
Checking for function "memset": YES
Checking for function "socket": YES
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Dependency libsodium found: NO
Header <stdbool.h> has symbol "_Bool": YES
Dependency threads found: YES
Configuring platform.hpp using configuration

Meson encountered an error in file 3rd/zeromq-4.1.6/meson.build, line 196, column 0:
Sandbox violation: Tried to grab file address.cpp from a different subproject.

A full log can be found at /home/maxgittelman/UxAS_pulls/OpenUxAS/build/meson-logs/meson-log.txt
```

Oddly if I `cd` into the zeromq directory the build succeeds.

```
~/UxAS_pulls/OpenUxAS/3rd/zeromq-4.1.6 $ meson build --buildtype=release
The Meson build system
Version: 0.43.0
Source dir: /home/maxgittelman/UxAS_pulls/OpenUxAS/3rd/zeromq-4.1.6
Build dir: /home/maxgittelman/UxAS_pulls/OpenUxAS/3rd/zeromq-4.1.6/build
Build type: native build
Project name: zeromq
Native C++ compiler: c++ (gcc 5.4.0)
Build machine cpu family: x86_64
Build machine cpu: x86_64
Has header "alloca.h": YES
Has header "arpa/inet.h": YES
Has header "dlfcn.h": YES
Has header "errno.h": YES
Has header "ifaddrs.h": YES
Has header "inttypes.h": YES
Has header "limits.h": YES
Has header "memory.h": YES
Has header "netinet/in.h": YES
Has header "netinet/tcp.h": YES
Has header "stdbool.h": YES
Has header "stddef.h": YES
Has header "stdint.h": YES
Has header "stdlib.h": YES
Has header "strings.h": YES
Has header "string.h": YES
Has header "sys/eventfd.h": YES
Has header "sys/socket.h": YES
Has header "sys/stat.h": YES
Has header "sys/time.h": YES
Has header "sys/types.h": YES
Has header "sys/uio.h": YES
Has header "time.h": YES
Has header "unistd.h": YES
Has header "windows.h": NO
Has header "sys/uio.h": YES
Checking if "kqueue links" links: NO
Checking if "epoll runs" runs: YES
Checking for function "clock_gettime": YES
Checking for function "fork": YES
Checking for function "gettimeofday": YES
Checking for function "gss_init_sec_context": NO
Checking for function "memset": YES
Checking for function "socket": YES
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Dependency libsodium found: NO
Header <stdbool.h> has symbol "_Bool": YES
Dependency threads found: YES
Configuring platform.hpp using configuration
Build targets in project: 1
Found ninja-1.7.2 at /usr/local/bin/ninja

```

Am I required to `cd` into this specific directory to build the project?

At the very top of the page, under "Quick Start" I cannot run your first simple test, and I'm hoping it has to do with the improper build. After running terminal 1 and hitting play, I get the following view:

![screenshot from 2017-10-09 13-36-01](https://user-images.githubusercontent.com/30596770/31350931-084266e4-acf7-11e7-802f-f3bd0e6483b7.png)

But in terminal 2, the following problem occurs:

```
~/UxAS_pulls/OpenUxAS/examples/02_Example_WaterwaySearch $ ./runUxAS_WaterwaySearch.sh 
rm: cannot remove './datawork': No such file or directory
rm: cannot remove './log': No such file or directory
./runUxAS_WaterwaySearch.sh: line 14: ../../../build/uxas: No such file or directory
```

**Comment by derekkingston** [comment]
Meson 0.43.0 breaks the UxAS build process with anticipated fixes in Meson 0.43.1. Current work around is to use Meson 0.42.1 which you can install from the git repo: https://github.com/mesonbuild/meson/releases/tag/0.42.1

**Comment by derekkingston** [comment]
Updated documentation to properly install correct meson version. Following steps should lead to correct compile on fresh Ubuntu system.
